modules/videoio/src/cap.cpp

```diff
@@ -364,56 +364,64 @@ CV_IMPL CvCapture * cvCreateCameraCapture (int index)
  * Videoreader dispatching method: it tries to find the first
  * API that can access a given filename.
  */
-CV_IMPL CvCapture * cvCreateFileCapture (const char * filename)
+CV_IMPL CvCapture * cvCreateFileCaptureWithPreference (const char * filename, int apiPreference)
 {
     CvCapture * result = 0;
 
+    switch(apiPreference) {
+    default:
+    case CV_CAP_FFMPEG:
 #ifdef HAVE_FFMPEG
-    if (! result)
-        result = cvCreateFileCapture_FFMPEG_proxy (filename);
+        if (! result)
```

Comment:
Watch out, right here and all additions below have the wrong identation. Please fix those to keep the code readable.